### DISCUSSION TITLE
Context in Field Resolvers

### DISCUSSION THREAD

**Discussion by jimbuck** [MichalLytek/type-graphql#896] [Q&A] [answered]
I'm trying to access the `@Ctx()` from within a FieldResolver, but it's always an empty object. Is there some other way to access it that I'm missing? I am using a custom decorator that wraps the built-in FieldResolver decorator (shown below). Is that causing the issue?

```ts
export function MyFieldResolver(...args: Parameters<typeof FieldResolver>): MethodDecorator {
  const field = FieldResolver(...args);
  const measurePerf = MeasurePerformance();

  // Combine the decorators into one decorator (both are void methods).
  return (target, property, descriptor) => {
    measurePerf(target, property, descriptor);
    field(target, property, descriptor);
  };
}
```

Thank you for any tips/insight you can provide!

**Comment by MichalLytek** [comment]
> Is that causing the issue?

There's only one way to check that!

1. Create a normal field resolver (with `@FieldResolver` only, without `MeasurePerformance`) and check if it can access the context.
2. Make sure you provide `context` in ApolloServer`.

**Reply by jimbuck** [reply]
Unfortunately just using `@FieldResolver` doesn't work either. For Apollo we have it setup like so:

```ts
context: ({ req }) => {
  const user = req?.user as User;
  return { user } as ApolloContext;
},
```

The odd thing is that it is working fine in mutations but just not the field resolvers. Is there a way to access the underlying request?

**Reply by MichalLytek** [reply]
>  Is there a way to access the underlying request?

Put `req` in the context.

> The odd thing is that it is working fine in mutations but just not the field resolvers. 

Please create a repository with a minimal reproducible code example.

**Reply by jimbuck** [reply]
The request is undefined in the context as well (but it's populated in the `context:` function I shared above). I will try to put an example repo together today. Thanks for the assistance!

**Comment by jimbuck** [comment, marked as answer]
So we actually figured out why the context was an empty object. The field resolver was being executed via the result of a subscription. Since it's not tied to one single request the context was empty, which is the correct behavior. We are updating our code to handle these scenarios.

Thanks for your help, and for the awesome library!
